subprojects/gradle/buildchecks/src/main/kotlin/com/avito/android/plugin/build_param_check/CheckTaskWithMode.kt

```diff
@@ -0,0 +1,12 @@
+package com.avito.android.plugin.build_param_check
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+
+internal abstract class CheckTaskWithMode : DefaultTask() {
```

Comment:
It looks like abstraction that we can inline

Reply:
Why? We want to make all checks to implement this mode.

Reply:
I don't get, when do we decide to make all checks implemented this mode?

Reply:
This proposal is made by @eugene-krivobokov on one of our daily sync (if my memory doesn't trick with me). So there is a task for that MBS-8390 (internal tracker). So I've made the foundation in the current task to make future works easy and not breaking backward compatibility.

Reply:
Yes, I'd like to migrate all checks from `enabled` flag to this enum.